File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int res = -1;
        int maxVal = INT_MIN;
        unordered_map<int, int> lvlSum;
        traverse(root, lvlSum, 1);
        for(auto& [lvl, val] : lvlSum)
        {
            if(val > maxVal)
            {
                res = lvl;
                maxVal = val;
            }
            else if(val == maxVal)
            {
                res = min(res, lvl);
            }
        }
        return res;
    }
private:
    void traverse(TreeNode* root, unordered_map<int, int>& lvlSum, int level)
    {
        if(!root)
        {
            return;
        }
        lvlSum[level] += root->val;
        traverse(root->left, lvlSum, level + 1);
        traverse(root->right, lvlSum, level + 1);
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
def traverse(root: Optional[TreeNode], lvlSum: dict[int, int], lvl: int) -> None:
    if root == None:
        return
    if lvl not in lvlSum:
        lvlSum[lvl] = 0
    lvlSum[lvl] += root.val
    traverse(root.left, lvlSum, lvl + 1)
    traverse(root.right, lvlSum, lvl + 1)
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        res = -1
        maxVal = float('-inf')
        lvlSum = {}
        traverse(root, lvlSum, 1)
        for lvl, val in lvlSum.items():
            if val > maxVal:
                res = lvl
                maxVal = val
            elif val == maxVal:
                res = min(res, lvl)
        return res
        